### PR TITLE
feat: Add `Snippets` and `SnippetPositions` functions

### PR DESCRIPTION
**Pull request by Cruel**
# Ticket(s) Closed

- Closes #

## What

Added `pdb.snippets` and `pdb.snippet_positions`.

## Why

Just adding features as I need them. 👍

### CODE REVIEW

File: tests/integration/test_snippet_functions.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        """Verify parameterized field check in Snippets.as_sql."""
+        snippets = Snippets("description")
+        snippets.source_expressions = [Value("parameterized")]
+        compiler = Mock()
+        compiler.compile.return_value = ("'parameterized'", ["parameterized"])
+        with pytest.raises(ValueError, match="does not support parameterized fields"):
+            snippets.as_sql(compiler, Mock())
+
+
+class TestSnippetIntegration:
```
[the remark this answers is not in the document]
Comment (Cruel) [reply tests/integration/test_snippet_functions.py]: I believe the second one was needed to fix existing lack of coverage:
https://app.codecov.io/gh/paradedb/django-paradedb/blob/main/src%2Fparadedb%2Ffunctions.py#L45

Image

But this is where I was unsure about how you want tests organized, especially as more things get implemented. Existing `Snippet` tests are in these files:
- `tests/test_edge_cases.py`
- `tests/integration/test_advanced_orm_integration.py`
- `tests/integration/test_paradedb_queries.py`
- `tests/integration/test_window_functions.py`

Should I go ahead and maintain this separation and add the appropriate category of tests to those files instead of adding this new file? (I personally tend to organize tests by feature, which why it made sense to add a new file)

File: tests/integration/test_snippet_functions.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    def test_snippets_invalid_sort_by_raises(self) -> None:
+        """Verify sort_by validation in integration suite."""
+        with pytest.raises(ValueError, match="sort_by must be one of"):
+            Snippets("description", sort_by="invalid")  # type: ignore[arg-type]
+
+    def test_snippets_parameterized_field_raises(self) -> None:
+        """Verify parameterized field check in Snippets.as_sql."""
+        snippets = Snippets("description")
+        snippets.source_expressions = [Value("parameterized")]
+        compiler = Mock()
```
[the remark this answers is not in the document]
Comment (Cruel) [reply tests/integration/test_snippet_functions.py]: That "parameterized" error check is more of an edge case. I'll move it to the edge case unit test file.

File: tests/integration/test_snippet_functions.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    """Integration tests for pdb.snippet_positions() function."""
+
+    def test_snippet_positions_basic(self) -> None:
+        """Verify pdb.snippet_positions returns byte offset pairs."""
+        qs = MockItem.objects.filter(description=ParadeDB("wireless")).annotate(
+            positions=SnippetPositions("description")
+        )
+        # "Innovative wireless earbuds" -> "wireless" starts at 11, ends at 19
+        item = qs.get(id=12)
+        assert item.positions == [[11, 19]]
```
[the remark this answers is not in the document]
Comment (Cruel) [reply tests/integration/test_snippet_functions.py]: Okay, I did this.

File: tests/integration/test_snippet_functions.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        """Verify sort_by returns expected fragment arrays."""
+        # For a single short description, score/position usually return the same.
+        # This test ensures the parameter is accepted and doesn't crash.
+        qs = MockItem.objects.filter(description=ParadeDB("shoes")).annotate(
+            fragments=Snippets("description", sort_by="score")
+        )
+        item = qs.first()
+        assert len(item.fragments) > 0
+        assert "<b>" in item.fragments[0]
+
```
Comment (Cruel) [inline tests/integration/test_snippet_functions.py]: Not sure how to test `sort_by` as the mock data is too small have multiple snippets to be sorted in any meaningful way.

**Comment by Cruel** [comment]
Okay @ankitml, I removed `TestSnippetIntegration` and kept the new file `test_snippet_functions.py` for now. Moved the few non-integration tests to the edge case file (the unit tests were pretty must edge cases anyways). Integration tests should be better now I think.

File: tests/integration/test_snippet_functions.py (hunk 4 above)
Comment (ankitml) [reply tests/integration/test_snippet_functions.py]: merging this then I have an update for test here https://github.com/paradedb/django-paradedb/commit/afc1f774dda46eaa5e3bc9d37ba4050711b07df4
